dcplib/etl/__init__.py

```diff
@@ -109,14 +122,14 @@ def enqueue_bundle_manifests(self, percent_complete, bundles_complete, total_bun
         return len(f2f_futures)
 
     def extract(self, query: dict = None, transformer: callable = None, loader: callable = None,
-                finalizer: callable = None, max_workers=512):
+                finalizer: callable = None, max_workers=512, executor_class=concurrent.futures.ThreadPoolExecutor):
         if query is None:
             query = self.default_bundle_query
         os.makedirs(f"{self.sd}/files", exist_ok=True)
         os.makedirs(f"{self.sd}/bundles", exist_ok=True)
         f2f_futures, ff_futures = [], []
         bundles_complete, percent_complete = 0, -1
-        with concurrent.futures.ThreadPoolExecutor(max_workers=max_workers) as executor:
+        with executor_class(max_workers=max_workers) as executor:
```

Comment:
Should this option should be made available for the callback instead of the downloading of bundles?